ui/app/src/components/Header/SearchBar.tsx

```diff
@@ -0,0 +1,223 @@
+// Copyright 2024 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { Box, Button, Chip, InputAdornment, Modal, Paper, TextField, Typography } from '@mui/material';
+import Magnify from 'mdi-material-ui/Magnify';
+import ViewDashboardIcon from 'mdi-material-ui/ViewDashboard';
+import Archive from 'mdi-material-ui/Archive';
+import React, { useMemo } from 'react';
+import { DashboardResource, ProjectResource } from '@perses-dev/core';
+import { KVSearch, KVSearchConfiguration, KVSearchResult } from '@nexucis/kvsearch';
+import { Link as RouterLink } from 'react-router-dom';
+import { useIsMobileSize } from '../../utils/browser-size';
+import { isAppleDevice } from '../../utils/os';
+import { useDashboardList } from '../../model/dashboard-client';
+import { useProjectList } from '../../model/project-client';
+
+const kvSearchConfig = {
+  indexedKeys: [['metadata', 'name']],
+  shouldSort: true,
+  includeMatches: true,
+} as KVSearchConfiguration;
+
+const sizeList = 10;
+
+function shortcutCTRL() {
```

Comment:
Shortcut not working on my Mac with Firefox :sad:

Reply:
yeah that's because it's not yet implemented, just the visualisation here